include/mqt-core/dd/Edge.hpp

```diff
@@ -88,6 +88,16 @@ template <class Node> struct Edge {
     return Edge{Node::getTerminal(), w};
   }
 
+  /**
+   * @brief Check whether an edge requires tracking.
+   * @param e The edge to check.
+   * @return Whether the edge requires tracking.
+   */
+  [[nodiscard]] static constexpr bool trackingRequired(const Edge& e) {
+    return !(e.isTerminal() && constants::isStaticNumber(e.w.r) &&
+             constants::isStaticNumber(e.w.i));
```

Comment:
```suggestion
    return !e.isTerminal() || !constants::isStaticNumber(e.w.r) ||
             !constants::isStaticNumber(e.w.i);
```
Might as well apply DeMorgan here directly.